modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/trigger/messaging/ObjectActionUserLoginTriggerMessageListener.java

```diff
@@ -0,0 +1,127 @@
+import com.liferay.object.system.SystemObjectDefinitionManagerRegistry;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.messaging.BaseMessageListener;
+import com.liferay.portal.kernel.messaging.Destination;
+import com.liferay.portal.kernel.messaging.DestinationConfiguration;
+import com.liferay.portal.kernel.messaging.DestinationFactory;
+import com.liferay.portal.kernel.messaging.DestinationNames;
+import com.liferay.portal.kernel.messaging.Message;
+import com.liferay.portal.kernel.messaging.MessageListener;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.util.MapUtil;
+import com.liferay.portal.vulcan.dto.converter.DTOConverterRegistry;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Nathaly Gomes
+ */
+@Component(
+	property = "destination.name=" + DestinationNames.USER_LOGIN,
+	service = MessageListener.class
+)
+public class ObjectActionUserLoginTriggerMessageListener
+	extends BaseMessageListener {
+
+	@Activate
+	protected void activate(BundleContext bundleContext) {
+		DestinationConfiguration destinationConfiguration =
+			new DestinationConfiguration(
+				DestinationConfiguration.DESTINATION_TYPE_SYNCHRONOUS,
+				DestinationNames.USER_LOGIN);
+
+		Destination destination = _destinationFactory.createDestination(
+			destinationConfiguration);
+
+		_serviceRegistration = bundleContext.registerService(
+			Destination.class, destination,
+			MapUtil.singletonDictionary(
+				"destination.name", destination.getName()));
+	}
+
+	@Deactivate
+	protected void deactivate() {
+		_serviceRegistration.unregister();
+	}
+
+	@Override
+	protected void doReceive(Message message) throws Exception {
+		long companyId = message.getLong("companyId");
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.fetchObjectDefinitionByClassName(
+				companyId, User.class.getName());
+
+		if (objectDefinition == null) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(
+					"User object definition is null for company " + companyId);
+			}
+
+			return;
+		}
+
+		User user = _userLocalService.getUser(message.getLong("userId"));
+
+		_objectActionEngine.executeObjectActions(
+			objectDefinition.getClassName(), message.getLong("companyId"),
+			ObjectActionTriggerConstants.KEY_ON_AFTER_LOGIN,
+			() -> ObjectEntryUtil.getPayloadJSONObject(
+				user, _dtoConverterRegistry, _jsonFactory,
+				ObjectActionTriggerConstants.KEY_ON_AFTER_LOGIN,
+				objectDefinition, null,
```

Comment:
Which payload are we passing to the action? I think the customer will need all the information related to the User (model or dto)